### PR TITLE
Add a Param example for precedence

### PR DESCRIPTION
**Pull request by maximlt**
As discussed on Discourse [here](https://discourse.holoviz.org/t/a-little-app-to-understand-how-precedence-works/378/4), here is the notebook for a Param example that demonstrates how the precedence parameter affects the display of widgets.

(It is my first PR ever so I might have done things completely wrong, let me know if that is the case!)

### CODE REVIEW

**Comment by jbednar** [comment]
That looks really helpful, thanks!  There's some metadata in there that we usually clear out, but we can do that after merging, so I think it looks fine.

File: examples/gallery/param/precedence.ipynb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    "> ``precedence`` is a value, usually in the range 0.0 to 1.0, that allows the order of Parameters in a class to be defined (for e.g. in GUI menus).  \n",
+    "> A negative precedence indicates a parameter that should be hidden in e.g. GUI menus.\n",
+    "\n",
+    "A `Param` pane has a ``display_threshold`` attribute defaulting to 0 and defined as follows:\n",
+    "\n",
+    "> Parameters with precedence below this value are not displayed.\n",
+    "\n",
+    "The interactive example below helps to understand how the interplay between these two attributes affects the display of widgets.\n",
+    "\n",
+    "The ``PrecedenceTutorial`` class emulates a dummy app whose display we want to control and that consists of three input parameters, ``x``, ``y`` and ``z``. These parameters will be displayed by `panel` with their associated default widgets. Additionnaly, the class declares the four parameters that will control the dummy app display: ``x_precedence``, ``y_precedence`` and ``z_precedence`` and ``dummy_app_display_threshold``."
```
Comment (jbednar) [inline examples/gallery/param/precedence.ipynb]:
```suggestion
    "The ``PrecedenceTutorial`` class emulates a dummy app whose display we want to control and that consists of three input parameters, ``x``, ``y`` and ``z``. These parameters will be displayed by `panel` with their associated default widgets. Additionally, the class declares the four parameters that will control the dummy app display: ``x_precedence``, ``y_precedence`` and ``z_precedence`` and ``dummy_app_display_threshold``."
```

**Comment by maximlt** [comment]
Thanks @jbednar for the review and comments. I thought about clearing the output as I saw in another pull request somewhere that it was required. I didn't think about all the metadata that a notebook can contain! Any easy way to strip that metadata out?

That might be a good opportunity to update the developer guide of panel to include some info about how to contribute to the docs. I could open an issue if you find this relevant.

**Comment by philippjfr** [comment]
>That might be a good opportunity to update the developer guide of panel to include some info about how to contribute to the docs. I could open an issue if you find this relevant.

Please do, for now I'm happy to merge. I regularly clean out the metadata before a release but if we add a docs section to the developer guide I'd be happy to add the notebook clearing script we use.

**Comment by philippjfr** [comment]
Thanks for contributing @maximlt! Much appreciated.

**Pull request merged**